packages/eslint-plugin/src/rules/unified-signatures.ts

```diff
@@ -215,6 +215,7 @@ export default util.createRule<Options, MessageIds>({
       if (ignoreDifferentlyNamedParameters) {
         for (let i = 0; i < a.params.length; i += 1) {
           if (
+            b.params[i] &&
```

Comment:
[Refactor] Asking for an element out of bounds of an array is a little confusing. It looks like this code is asking whether `b.params[i]` is a falsy element - even though the type of `b.params` indicates that not to be doable. I'm also under the impression accessing elements out of an array's bounds can cause a performance deoptimization (though this code almost certainly wouldn't suffer meaningfully).

I think a more proper fix would be capping the `i` to `Math.min(a.params.length, b.params.length)`.

Reply:
Just applied - normally I'd let you have the pleasure, but I feel bad taking up your time on such a small change 😄